ros2bag/ros2bag/verb/record.py

```diff
@@ -242,7 +242,7 @@ def check_necessary_argument(args):
 def validate_parsed_arguments(args, uri) -> str:
     if args.topics_positional:
         print(print_warn('Positional "topics" argument deprecated. '
-                         'Please use optional "--topics" argument instead.'))
+                         'Please use optional "--topics" argument instead.'), flush=True)
```

Comment:
We usually set the `PYTHONUNBUFFERED` env var to `1` for tests to get this behaviour. This way we don't need to flush (which can be expensive) for normal uses of the tool.

Could we instead set `PYTHONUNBUFFERED=1` in the relevant tests?

Reply:
> Could we instead set PYTHONUNBUFFERED=1 in the relevant tests?

I don't think so. The failure is happening in the integration tests when we are runing `ros2 bag record` in a separate process from the C++ code.

Reply:
You can set `PYTHONUNBUFFERED=1` in the child process after forking here, but before executing the command: https://github.com/ros2/rosbag2/blob/dcaca47599611a378d20fc897fdfc14df8813c92/rosbag2_test_common/include/rosbag2_test_common/process_execution_helpers_unix.hpp#L83-L96

If you use [`execvpe()`](https://linux.die.net/man/3/execvpe), it should be easy to set an environment variable.

I'm suggesting this because you wouldn't have to set `flush=True` for _all_ prints, which seems more scalable to me. It would also only flush in tests, and not for normal uses. Your choice.

Reply:
Sorry, It might have other side affects for other tests if we will set `PYTHONUNBUFFERED=1` by default for any new process runing from tests.
I would prefer to fix the root cause of the problem in the source file rather than the consequences in the tests.